### ISSUE TITLE
called `Result::unwrap()` on an `Err` value: RelativeUrlWithCannotBeABaseBase

### ISSUE THREAD

**Issue by ConnorGBrewster** [servo/servo#12248]
Website: `./mach run -d -w https://maggieandcaleb.com/`
```
ERROR:constellation::constellation: Backtrace:
frame #0  - 0x0000000109fddafe - backtrace::backtrace::trace::hccde8df28b4db2a2
frame #1  - 0x0000000109fdda81 - backtrace::capture::Backtrace::new::h42f95930bb8c5ee8
frame #2  - 0x00000001092a14a6 - _<F as alloc..boxed..FnBox<A>>::call_box::hebb3b95724d7c6a0
frame #3  - 0x0000000109f4d6a7 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0fcde37d4d058867
frame #4  - 0x000000010a1c35b4 - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
frame #5  - 0x000000010a1dc4f6 - std::panicking::begin_panic::he426e15a3766089a
frame #6  - 0x000000010a1c4bf8 - std::panicking::begin_panic_fmt::hdddb415186c241e7
frame #7  - 0x000000010a1dc14f - rust_begin_unwind
frame #8  - 0x000000010a202fd0 - core::panicking::panic_fmt::hf4e16cb7f0d41a25
frame #9  - 0x00000001090903e9 - core::result::unwrap_failed::hf81255c6633560f3
frame #10 - 0x000000010913591c - _<script..dom..htmlimageelement..HTMLImageElement as script..dom..virtualmethods..VirtualMethods>::attribute_mutated::hc945bc8431150ee0
frame #11 - 0x00000001090c1678 - script::dom::element::Element::push_attribute::h256cc0f248f930a7
frame #12 - 0x00000001090c0f94 - script::dom::element::Element::push_new_attribute::ha51265302c277025
frame #13 - 0x00000001090c1c9b - script::dom::element::Element::set_attribute_from_parser::hf1f8d81b4b6561f4
frame #14 - 0x00000001091d78e2 - script::parse::html::_<impl html5ever..tree_builder..interface..TreeSink for script..dom..servohtmlparser..Sink>::create_element::hfe8a5ab6183b966a
frame #15 - 0x00000001091d7434 - _<html5ever..tree_builder..TreeBuilder<Handle, Sink> as html5ever..tree_builder..actions..TreeBuilderActions<Handle>>::insert_element::hdf8ac12b88b70e09
frame #16 - 0x00000001091ed478 - _<html5ever..tree_builder..TreeBuilder<Handle, Sink> as html5ever..tree_builder..rules..TreeBuilderStep>::step::hca68e2bc2432aa52
frame #17 - 0x00000001091cd5e0 - _<html5ever..tree_builder..TreeBuilder<Handle, Sink> as html5ever..tokenizer..interface..TokenSink>::process_token::he45af09b32dfea6d
frame #18 - 0x00000001091cc9a1 - _<html5ever..tokenizer..Tokenizer<Sink>>::process_token::hf3778c2010ac92bf
frame #19 - 0x000000010920db85 - _<html5ever..tokenizer..Tokenizer<Sink>>::emit_current_tag::h7e260b3f099be738
frame #20 - 0x000000010920944e - _<html5ever..tokenizer..Tokenizer<Sink>>::step::hc800f3fb49ad6b84
frame #21 - 0x00000001091faf47 - _<html5ever..tokenizer..Tokenizer<Sink>>::run::ha80cf9bcfaf349a8
frame #22 - 0x00000001091c864c - script::dom::servohtmlparser::ServoHTMLParser::parse_sync::ha88022f6608066ce
frame #23 - 0x00000001091c828d - _<&'a script..dom..servohtmlparser..ServoHTMLParser as script..parse..Parser>::parse_chunk::hae58eee26761b177
frame #24 - 0x00000001090a6ae8 - script::parse::html::parse_html::hde86e03b886f8d92
frame #25 - 0x000000010917f669 - script::parse::html::parse_html_fragment::hfb037293a8a02e5e
frame #26 - 0x00000001090c850b - script::dom::node::Node::parse_fragment::h1f21413fdc5021f0
frame #27 - 0x0000000108c97832 - _<script..dom..element..Element as script..dom..bindings..codegen..Bindings..ElementBinding..ElementMethods>::SetInnerHTML::h7ebcad5447a38e10
frame #28 - 0x0000000108c97665 - std::panicking::try::call::haabc8a04b7634478
frame #29 - 0x000000010a1df1db - __rust_try
frame #30 - 0x000000010a1df175 - __rust_maybe_catch_panic
frame #31 - 0x0000000108c97475 - script::dom::bindings::codegen::Bindings::ElementBinding::set_innerHTML::h65c92bf65fa4af4e
frame #32 - 0x00000001098b8c3a - CallJitSetterOp
frame #33 - 0x0000000108afe74d - script::dom::bindings::utils::call_setter::hda6b2e0134eb21e0
frame #34 - 0x0000000108afe60a - script::dom::bindings::utils::generic_call::h53d97b1b51a156c2
frame #35 - 0x0000000109cec3a0 - 2js6InvokeEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
frame #36 - 0x0000000109d05cf7 - 2js6InvokeEP9JSContextRKN2JS5ValueES5_jPS4_NS2_13MutableHandleIS3_E
frame #37 - 0x0000000109d064b8 - 2js12InvokeSetterEP9JSContextRKN2JS5ValueES3_NS2_6HandleIS3_E
frame #38 - 0x0000000109d1a9fb - 2js17NativeSetPropertyEP9JSContextN2JS6HandleIPNS_12NativeObjectEEENS3_I4jsidEENS3_INS2_5ValueEEESA_NS_13QualifiedBoolERNS2_14ObjectOpResult
frame #39 - 0x0000000109cfc15f - _ZL9InterpretP9JSContextRN2js8RunStateE
frame #40 - 0x0000000109cf4d4d - 2js9RunScriptEP9JSContextRNS_8RunState
frame #41 - 0x0000000109cec711 - 2js6InvokeEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
frame #42 - 0x0000000109c188a6 - 2js8fun_callEP9JSContextjPN2JS5Value
frame #43 - 0x0000000109cec3a0 - 2js6InvokeEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
frame #44 - 0x0000000109cff381 - _ZL9InterpretP9JSContextRN2js8RunStateE
frame #45 - 0x0000000109cf4d4d - 2js9RunScriptEP9JSContextRNS_8RunState
frame #46 - 0x0000000109cec711 - 2js6InvokeEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
frame #47 - 0x0000000109d05cf7 - 2js6InvokeEP9JSContextRKN2JS5ValueES5_jPS4_NS2_13MutableHandleIS3_E
frame #48 - 0x0000000109beb699 - _Z20JS_CallFunctionValueP9JSContextN2JS6HandleIP8JSObjectEENS2_INS1_5ValueEEERKNS1_16HandleValueArrayENS1_13MutableHandleIS6_EE
frame #49 - 0x0000000108cab62e - script::dom::bindings::codegen::Bindings::EventListenerBinding::EventListener::HandleEvent::hb9dec6c7653f07ef
frame #50 - 0x00000001090e5ab5 - script::dom::eventdispatcher::invoke::ha664c8406df4c096
frame #51 - 0x00000001090e859a - script::dom::eventdispatcher::dispatch_event::h2af7a9aca7ed0603
frame #52 - 0x000000010927f6ff - script::dom::xmlhttprequest::XMLHttpRequest::change_ready_state::he491938184f42e35
frame #53 - 0x0000000109280ec3 - script::dom::xmlhttprequest::XMLHttpRequest::process_partial_response::hadcbbfa4b87a46d5
frame #54 - 0x000000010927ec08 - script::dom::xmlhttprequest::XMLHttpRequest::initiate_async_xhr::_<impl net_traits..FetchResponseListener for script..dom..xmlhttprequest..XHRContext>::process_response_eof::hd6f9be77534fad08
frame #55 - 0x000000010927cf38 - _<script..network_listener..ListenerRunnable<A, Listener> as script..script_thread..Runnable>::handler::he876d76c659abc96
frame #56 - 0x00000001092acd7c - script::script_thread::ScriptThread::handle_msg_from_script::h6a6b4ae37d32650d
frame #57 - 0x0000000109308e91 - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::hdfba591677ee0854
frame #58 - 0x00000001092eeb5f - script::script_thread::ScriptThread::handle_msgs::h45a47f1f1786f911
frame #59 - 0x000000010929fd27 - std::panicking::try::call::h78061d8f9ce27ca7
frame #60 - 0x000000010a1df1db - __rust_try
frame #61 - 0x000000010a1df175 - __rust_maybe_catch_panic
frame #62 - 0x00000001092a0ed4 - _<F as alloc..boxed..FnBox<A>>::call_box::h2ec158fb92e41aca
frame #63 - 0x000000010a1dac58 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
frame #64 - 0x00007fff9b96f99c - _pthread_body
frame #65 - 0x00007fff9b96f919 - _pthread_start
```

**Comment by jdm** [comment]
Looks like the same problem that causes #11992. Should be fixed by #12003.

**Comment by jdm** [comment]
Thank you for reporting it ;)
